### PR TITLE
conditionally add `-Wno-unused-command-line-argument` to `$CFLAGS` to fix error when installing `imkl-FFTW` with RPATH

### PR DESCRIPTION
**Pull request by casparvl**
Fix for https://github.com/easybuilders/easybuild-easyconfigs/pull/18439#issuecomment-1662671054 which pops up as:

```
icx: error: -Wl,-rpath=$ORIGIN/../lib64: 'linker' input unused [-Werror,-Wunused-command-line-argument]
icx: error: -Wl,--disable-new-dtags: 'linker' input unused [-Werror,-Wunused-command-line-argument]
```

### CODE REVIEW

**Comment by casparvl** [comment]
Test report by @casparvl
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _imkl-FFTW-2022.1.0-iimpi-2022a.eb_ 
 * **SUCCESS** _imkl-FFTW-2021.4.0-iimpi-2021b.eb_ 
 * **SUCCESS** _imkl-2023.1.0.eb_ 
 * **SUCCESS** _imkl-2023.0.0.eb_ 
 * **SUCCESS** _imkl-2022.2.1.eb_ 
 * **SUCCESS** _imkl-2022.2.0.eb_ 
 * **SUCCESS** _imkl-FFTW-2022.2.1-iimpi-2022b.eb_ 
 * **SUCCESS** _iimpi-2022.11.eb_ 
 * **SUCCESS** _imkl-FFTW-2022.2.1-iimpi-2022.11.eb_ 
 * **SUCCESS** _intel-compilers-2023.1.0.eb_ 
 * **SUCCESS** _intel-compilers-2023.0.0.eb_ 
 * **FAIL (build issue)** _intel-compilers-2022.2.0.eb_ (partial log available at https://gist.github.com/casparvl/90918f7808cb4c8545f20cfab06dea60)
 * **FAIL (build issue)** _impi-2021.9.0-intel-compilers-2023.1.0.eb_ (partial log available at https://gist.github.com/casparvl/5070bed9add1aa03e5e6ea34ad2e327e)
 * **FAIL (build issue)** _impi-2021.8.0-intel-compilers-2023.0.0.eb_ (partial log available at https://gist.github.com/casparvl/7c4ba28f63ff2a9a4346b9bf729c5ed4)
 * **FAIL (build issue)** _impi-2021.7.0-intel-compilers-2022.2.0.eb_ (partial log available at https://gist.github.com/casparvl/8c6571ea25404e40739dd066b21724eb)
 * **FAIL (build issue)** _iimpi-2023a.eb_ (partial log available at https://gist.github.com/casparvl/c0b1d8e29f14f24a574213214879c357)
 * **FAIL (build issue)** _iimpi-2023.03.eb_ (partial log available at https://gist.github.com/casparvl/6b55394aca0a4fd80c2899db63833795)
 * **FAIL (build issue)** _iimpi-2022.12.eb_ (partial log available at https://gist.github.com/casparvl/6c1e2098e3abe2eb5d818bc676c97d62)
 * **FAIL (build issue)** _iimpi-2022.09.eb_ (partial log available at https://gist.github.com/casparvl/1474dfafa85a06cdc2eb3bb7110f2439)
 * **FAIL (build issue)** _imkl-FFTW-2023.1.0-iimpi-2023a.eb_ (partial log available at https://gist.github.com/casparvl/57e962693408879d098dbacc106570bc)
 * **FAIL (build issue)** _imkl-FFTW-2023.1.0-iimpi-2023.03.eb_ (partial log available at https://gist.github.com/casparvl/588ffe0e0a1d35d6b56086c7fc2aa98e)
 * **FAIL (build issue)** _imkl-FFTW-2023.0.0-iimpi-2022.12.eb_ (partial log available at https://gist.github.com/casparvl/88473742073294387a98744e7d7114ed)
 * **FAIL (build issue)** _imkl-FFTW-2022.2.0-iimpi-2022.09.eb_ (partial log available at https://gist.github.com/casparvl/6ab5f8baecf1497fb948e971e49a0a5e)

Build succeeded for 11 out of 23 (8 easyconfigs in total)
tcn174.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/db7d09a93a3330968889beab080dac2d for a full test report.

**Comment by casparvl** [comment]
Ugh, I'm getting failures because I'm exceeding my inode quotum... Will cleanup first, then retry...

File: easybuild/easyblocks/i/imkl.py [hunk 1]
```diff
@@ -261,14 +262,21 @@ def build_mkl_fftw_interfaces(self, libdir):
                 tmpbuild = tempfile.mkdtemp(dir=self.builddir)
                 self.log.debug("Created temporary directory %s" % tmpbuild)
 
+                # Avoid unused command line arguments (-Wl,rpath...) causeing errors when using RPATH
+                # See https://github.com/easybuilders/easybuild-easyconfigs/pull/18439#issuecomment-1662671054
+                if build_option('rpath'):
+                    cflags = flags + ' -Wno-unused-command-line-argument'
```
Comment (boegel) [inline easybuild/easyblocks/i/imkl.py:268]: This option is probably only supported for Intel oneAPI (Clang-based) compilers, so this needs an extra condition?

Comment (casparvl) [reply easybuild/easyblocks/i/imkl.py:268]: Had some discussion on Slack. This is correct: it's supported by Clang-based compilers. Hence we should limit it to those. The problem that I'm trying to address in this PR should also only occur for Clang based compilers (that's why we haven't run into it with prior versions if imkl-fftw: we didn't use the oneAPI compilers yet by default there).

**Comment by bedroge** [comment]
Looks like the same issue as this one: https://github.com/easybuilders/easybuild-easyblocks/issues/2910 (which I tried to fix with https://github.com/easybuilders/easybuild-easyblocks/pull/2912).

**Comment by casparvl** [comment]
Test report by @casparvl
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _imkl-FFTW-2022.2.1-iimpi-2022b.eb_ 
 * **SUCCESS** _imkl-FFTW-2022.2.1-iimpi-2022.11.eb_ 
 * **SUCCESS** _imkl-FFTW-2022.1.0-iimpi-2022a.eb_ 
 * **SUCCESS** _imkl-FFTW-2021.4.0-iimpi-2021b.eb_ 
 * **SUCCESS** _impi-2021.9.0-intel-compilers-2023.1.0.eb_ 
 * **SUCCESS** _impi-2021.8.0-intel-compilers-2023.0.0.eb_ 
 * **SUCCESS** _intel-compilers-2022.2.0.eb_ 
 * **SUCCESS** _iimpi-2023a.eb_ 
 * **SUCCESS** _iimpi-2023.03.eb_ 
 * **SUCCESS** _iimpi-2022.12.eb_ 
 * **SUCCESS** _imkl-FFTW-2023.1.0-iimpi-2023a.eb_ 
 * **SUCCESS** _imkl-FFTW-2023.1.0-iimpi-2023.03.eb_ 
 * **SUCCESS** _imkl-FFTW-2023.0.0-iimpi-2022.12.eb_ 
 * **SUCCESS** _impi-2021.7.0-intel-compilers-2022.2.0.eb_ 
 * **SUCCESS** _iimpi-2022.09.eb_ 
 * **SUCCESS** _imkl-FFTW-2022.2.0-iimpi-2022.09.eb_ 

Build succeeded for 16 out of 16 (8 easyconfigs in total)
tcn174.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/1520e3ad48145b0e1a58388bafff2a0f for a full test report.

**Comment by casparvl** [comment]
> Looks like the same issue as this one: #2910 (which I tried to fix with #2912).

Arrggg, why do I have such a terrible memory, I even commented in that issue... 

So, @bedroge @boegel any preference? @bedroge 's approach is a bit more targetted. That has the advantage that we _know_ this change has to be made for `intel-compilers` when using the `oneapi` compilers. My PR would also include `clang` compiling `FFTW`. That would _probably_ work, but is untested. One could argue that it is easy enough to add `clang` to the `if`-statement in @bedroge 's PR once it is relevant.

The downside is there was some issue with `--read-only-installdir` it seems, in @bedroge 's PR? Though @boegel wasn't sure.

File: easybuild/easyblocks/i/imkl.py [hunk 2]
```diff
@@ -261,14 +262,21 @@ def build_mkl_fftw_interfaces(self, libdir):
                 tmpbuild = tempfile.mkdtemp(dir=self.builddir)
                 self.log.debug("Created temporary directory %s" % tmpbuild)
 
+                # Avoid unused command line arguments (-Wl,rpath...) causeing errors when using RPATH
```
Comment (boegel) [inline easybuild/easyblocks/i/imkl.py:265]:
```suggestion
                # Avoid unused command line arguments (-Wl,rpath...) causing errors when using RPATH

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
The approach takes in https://github.com/easybuilders/easybuild-easyblocks/pull/2912 highlight a problem with `imkl-FFTW` which we should fix first, but the changes here are OK short-term (it doesn't make a bad situation worse, and it's a reasonable fix for the problem), so let's merge this PR, and revisit https://github.com/easybuilders/easybuild-easyblocks/pull/2912 once `imkl-FFTW` has been changed to not build the FFTW interfaces directly in the `imkl` installation directory.

**Review by boegel — APPROVED** [review]

**Pull request merged**